db/migrate/20170414140927_create_incoming_message_error.rb

```diff
@@ -0,0 +1,10 @@
+# -*- encoding : utf-8 -*-
+class CreateIncomingMessageError < ActiveRecord::Migration
+  def change
+    create_table :incoming_message_errors do |t|
+      t.timestamps null: false
+      t.text :unique_id, null: false
+      t.datetime :retry_at
```

Comment:
I'd be inclined to store the exception message/stack trace here to give a bit of context around what the error was. I'd always have to look up the difference between a `retry_at` existing and not existing, so having the error alongside would be a good sanity check. Since we also rescue `StandardError`, something unexpected might end up here, so the extra context would be good in that case.